File: backend-api/src/models/Review.js
const mongoose = require('mongoose');

const ReviewSchema = new mongoose.Schema({
  restaurantId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Restaurant',
    required: true
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  userName: { type: String, required: true },
  userAvatarUrl: { type: String, default: '' },
  rating: { type: Number, required: true, min: 1, max: 5 },
  comment: { type: String, required: true }
}, {
  timestamps: true // Esto creará automáticamente createdAt y updatedAt
});

module.exports = mongoose.model('Review', ReviewSchema);

File: android-app/app/src/main/java/com/example/foodboxd/domain/model/Review.kt
package com.example.foodboxd.domain.model

data class Review(
    val id: String,
    val restaurantId: String,
    val userId: String,
    val userName: String,
    val userAvatarUrl: String,
    val rating: Double,
    val comment: String,
    val date: String // e.g. "2026-06-02" or friendly date
)
